frontend/src/stores/upload.ts

```diff
@@ -47,6 +47,41 @@ export const useUploadStore = defineStore("upload", {
       const sum = state.progress.reduce((acc, val) => +acc + +val) as number;
       return Math.ceil((sum / totalSize) * 100);
     },
+    getProgressDecimal: (state) => {
+      if (state.progress.length === 0) {
+        return 0;
+      }
+
+      const totalSize = state.sizes.reduce((a, b) => a + b, 0);
+
+      // TODO: this looks ugly but it works with ts now
+      const sum = state.progress.reduce((acc, val) => +acc + +val) as number;
```

Comment:
Why `+` sign is used before `acc` and `val`?

Reply:
@kloon15 implemented the original logic and could probably explain better than I could, but that's a use of the [unary operator.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus)

Basically, the + sign in front of the acc and val turns them into numbers, and ensures that we are adding numbers and not concatenating strings.

Reply:
Pretty much what @alexjyong said; the problem is that boolean is also assigned to the same variable, which im not sure why and how its used elsewhere and to not break anything while converting to ts, it was just kept as is hence the explicit casting to number.